src/foam/core/reflow/UploadAgent.js

```diff
@@ -48,28 +59,59 @@ foam.CLASS({
 
             // Deserialize the decompressed data back to FObject array
             String decompressedJson = new String(baos.toByteArray(), "UTF-8");
+
             foam.lib.json.JSONParser parser = new foam.lib.json.JSONParser();
             parser.setX(getX());
-            Object[] arrayResult = parser.parseStringForArray(decompressedJson, null);
 
-            if ( arrayResult != null && arrayResult.length > 0 ) {
+            // Parse the JSON - returns null on error (doesn't throw by default)
+            Object[] arrayResult = null;
+            try {
+              arrayResult = parser.parseStringForArray(decompressedJson, null);
+            } catch (RuntimeException t) {
+              // Parsing threw an exception - get detailed error like ServiceWebAgent
+              try {
+                String detailedError = getParsingError(getX(), decompressedJson);
+                logger.error("UploadAgent", "JSON parsing exception", "error", detailedError);
```

Comment:
```suggestion
                logger.error("UploadAgent", "JSON parsing exception", detailedError);
```
'error' is redundant. An error log output will already contain the 'error' string.